databuilder/backends/tpp.py

```diff
@@ -249,21 +249,42 @@ class TPPBackend(BaseBackend):
                 ethnic___9,
                 ethnic___10,
                 covid19_vaccine,
-                CAST(covid19_vaccined AS DATE) as covid19_vaccined,
-                CAST(covid19_vaccine2d AS DATE) as covid19_vaccine2d,
+                CASE
+                    WHEN covid19_vaccined = 'NA' THEN null
+                    ELSE CAST(covid19_vaccined AS DATE)
```

Comment:
Sadly, and there's no way you could have known this, you can't just cast strings to dates because the way MSSQL choose to parse them is locale dependent. You might think that ISO dates are unambiguous, and they are. But (utterly bizarrely) MSSQL can sometimes parse them as `YYYY-DD-MM` — a sort of ungodly hybrid US/ISO style which exists nowhere else in the world.

The safe way to do this is to use `CONVERT` and explicitly specify the format (which for ISO format happens to be number 23).

More gory details here:
https://learn.microsoft.com/en-us/sql/t-sql/data-types/nondeterministic-convert-date-literals?view=sql-server-ver16

```suggestion
                    ELSE CONVERT(DATE, covid19_vaccined, 23)
```

Reply:
FFS! Thanks, Dave.

Reply:
ouch! I'll fix this. (I'm fortunate enough to have not used MSSQL before.)